tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -254,14 +330,46 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
                            m::Convert(m::CustomCall(&existing_gemm,
                                                     {kGemmCallTarget,
                                                      kCublasLtMatmulCallTarget})
-                                          .WithElementType(BF16)),
-                           m::Convert(m::Op(&bias).WithElementType(BF16))))
+                                          .WithOneUser()
+                                          .WithElementType(BF16))
+                               .WithOneUser(),
+                           m::Convert(m::Op(&bias).WithElementType(BF16))
+                               .WithOneUser())
+                           .WithOneUser())
                 .WithElementType(BF16))) {
       return FuseMatrixBiasAdd(instr, bias, existing_gemm);
     }
     return OkStatus();
   }
 
+  Status SinkSliceBelowBinaryOp(HloInstruction *slice, HloInstruction **binary,
+                                HloInstruction **broadcast) {
+    // Replaces binary(slice(gemm), broadcast) with
+    // slice(binary(gemm, broadcast)) and changes the shape of broadcast from
+    // that of slice to that of the GEMM, i.e. the operand of slice.
```

Comment:
Thanks for the comment, it is great.

Nit: Move this comment to right above the function.  This is where we put comments that explain "what a function does" like this one.